core/jobs/batch_jobs/reject_invalid_suggestion_and_delete_invalid_translation_jobs.py

```diff
@@ -0,0 +1,468 @@
+                strict=True,
+                version=entity_translation_model.entity_version)
+        exp = exp_fetchers.get_exploration_from_model(exp_model)
+
+        exp_content_ids = exp.get_translatable_content_ids()
+        translated_content_ids = list(
+            entity_translation_model.translations.keys())
+
+        deleted_translations_count = 0
+        is_updated = False
+        for content_id in translated_content_ids:
+            if content_id not in exp_content_ids:
+                entity_translation_model.translations.pop(content_id)
+                deleted_translations_count += 1
+                is_updated = True
+
+        if is_updated:
+            result: Dict[str, Union[
+                translation_models.EntityTranslationsModel, int]] = {
+                'entity_translation_model': entity_translation_model,
+                'deleted_translations_count': deleted_translations_count
+            }
+            return result
+        return None
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns a PCollection of entity translation model update results.
+
+        Returns:
+            PCollection. A PCollection of the job run results.
+        """
+        entity_translation_models = _get_entity_translation_models(
+            self.pipeline)
+        deletion_result_dicts = (
+            entity_translation_models
+            | 'Get deletion results' >> beam.Map(
+                    self._delete_translations_with_invalid_content_ids)
+            | 'Filter out None values' >> beam.Filter(lambda x: x is not None)
+        )
+
+        deleted_translations_count_job_run_results = (
+            deletion_result_dicts
+            | 'Deleted translations counts' >> beam.Map(
+                    lambda x: x['deleted_translations_count'])
+            | 'Total deleted translations count' >> (
+                beam.CombineGlobally(sum))
+            | 'Only create result for non-zero number of objects' >> (
+                beam.Filter(lambda x: x > 0))
+            | 'Report total deleted translations count' >> beam.Map(
+                lambda result: (
+                    job_run_result.JobRunResult.as_stdout(
+                        f'DELETED TRANSLATIONS COUNT SUCCESS: {result}'
+                    )))
+        )
+
+        updated_entity_translation_models = (
+            deletion_result_dicts
+            | 'Updated entity translation models' >> beam.Map(
+                    lambda x: x['entity_translation_model'])
+        )
+
+        updated_entity_translation_models_count_job_run_results = (
+            updated_entity_translation_models
+            | 'Updated entity transltion models count' >> (
+                job_result_transforms.CountObjectsToJobRunResult(
+                    'UPDATED ENTITY TRANSLATION MODELS COUNT'))
+        )
+
+        unused_put_results = (
+            updated_entity_translation_models
```

Comment:
I think after updating entity translations, we need to update translation_counts in the opportunity models as well?

Reply:
Thanks @chris7716 for highlighting it, will make the necessary changes.

Reply:
Hey @chris7716, I am sorry for the delay in adding changes. PTAL

Reply:
Thanks for adding it.